nicos/ess-nicos-roadmap.md

```diff
@@ -0,0 +1,78 @@
+ - does everybody have all the accounts they need git/gerrit/slack...? ("get started" page for new developers or similar?)
+ - DMSC contribution to infrastructure? build servers? ESS artifactory interesting for hosting packages?
+ - issues will be tracked in redmine - ESS tag/separate area?
+ - participation in meetings/interaction
+    * ESS ECP workshops every 6 month - interesting?
+    * weekly/monthly? developer meetings - video conferencing?
+    * frequency for face-to-face meetings? organize working in same place (code camp or similar) if time allows?
+ - inclusion of ESS staff in general maintenance tasks (contributing effort for non ESS-related issues)
+ - ensure knowledge of ESS instrument control "lives in NICOS" (documentation etc) (important for staff turnover)
+    
+ESS specific developments
+-------------------------
+ - Similar to custom/frm2
+ - Should contain ESS specific generic setups/device types
+    * Example: EPICS motor record is totally generic, ESS motors will have some extensions (additional PVs)
+    * Pending final (single disk) chopper EPICS interface, integrate.
+
+
+EPICS
+-----
+ - improve support for EPICS ChannelAccess devices using pyepics
+    * better default implementation of doStatus (using EPICS alarm states)
+    * implement callback-based poller that monitors PVs and updates cache
+    * improve EpicsMotor and move into "core" system
+
+ - support pvAccess using pvaPy module
+    * pending use cases from ICS/DM beyond "plain PVs"
+    * should eventually also replace pyepics for ChannelAccess
+
+
+Integration with other DMSC infrastructure
+------------------------------------------
+ - Deployment strategy/implementation (mainly with DST group, possibly ICS)
+    * compatibility of server components (daemon, cache, poller, ...) with Centos 7.1 (rpm packages, analoguous to deb)
+    * automated server deployment & configuration (environment variables to select correct instrument
+      on specific machine etc)
+    * compatibility of client components (mainly GUI) with Centos 7.1 and Mac OS X (likely used by many ESS beamline scientists)
+    * installation of client and potentially server in EEE development environment (also Centos 7.1)
+    * driven forward as part of LDPC (live data prototype collaboration)
+
+ - Control & query the DM group's aggregator (what gets aggregated), file writing, ...
+    * interface to detector data for all ESS instruments
+    * pending exact protocol/mechanism, prototyping at V20
+    * distribution of responsibility between NICOS/aggregator (filenames, ...)
+    * potentially aggregate quantities derived by NICOS (wavelenghts, ...)
+    * also driven forward in LDPC
+
+ - Integration with Mantid
+    * InstrumentView-widget as first step for visualisation
+    * towards data processing scripts (either on local machine or on remote - depends on data streaming/file writing?)
+
```

Comment:
I'd say this is a very specific problem for which we don't have a clear (user-requested) use case for. My suggestion would be to wait until we have a better understanding of what different instruments would need for that.